opal/mca/btl/uct/btl_uct_tl.c

```diff
@@ -461,8 +461,12 @@ static void mca_btl_uct_set_tl_am (mca_btl_uct_module_t *module, mca_btl_uct_tl_
 	tl->max_device_contexts = mca_btl_uct_component.num_contexts_per_module;
     }
 
-    module->super.btl_max_send_size = MCA_BTL_UCT_TL_ATTR(tl, 0).cap.am.max_zcopy - sizeof (mca_btl_uct_am_header_t);
     module->super.btl_eager_limit = MCA_BTL_UCT_TL_ATTR(tl, 0).cap.am.max_bcopy - sizeof (mca_btl_uct_am_header_t);
+    if (MCA_BTL_UCT_TL_ATTR(tl, 0).cap.am.max_zcopy) {
```

Comment:
according to UCT spec, to check for AM_ZCOPY support , need to check cap.flags for UCT_IFACE_FLAG_AM_ZCOPY, rather than cap.am.max_zcopy.

Reply:
fixed